### PR TITLE
[MRESOLVER-56] Support SHA-256 and SHA-512 as checksums

### PR DESCRIPTION
**Pull request by michael-o**

### CODE REVIEW

File: maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java [hunk 1]
```diff
@@ -8,9 +8,9 @@
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
- * 
+ *
```
Comment (mthmulders) [inline maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java:11]: Unrelated change

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java [hunk 2]
```diff
@@ -37,18 +37,32 @@
     {
         try
         {
-            digest = MessageDigest.getInstance( "SHA-1" );
+            digest = MessageDigest.getInstance( "SHA-512" );
```
Comment (mthmulders) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:40]: Can't we replace this fourfold nested try/catch with a loop over an ordered collection?

Comment (rfscholte) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:40]: I spotted the same codesmell: nested try-catch blocks. Should be able to rewrite that. with a single try-catch level

File: maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java (hunk 1 above)
Comment (michael-o) [reply maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java:11]: That's because my Eclipse autotrims trailing whitespace. I can undo that of course.

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java (hunk 2 above)
Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:40]: Will do and catch the last exception.

Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java]: Done!

File: maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java (hunk 1 above)
Comment (mthmulders) [reply maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java:11]: I've seen other pull requests against Maven where changing whitespace was not OK, so I figured that it wouldn't be OK here either.

Comment (michael-o) [reply maven-resolver-connector-basic/src/test/java/org/eclipse/aether/connector/basic/ChecksumCalculatorTest.java:11]: I wouldn't object if you provide a PR where trailing whitescape is trimmed. What we dislike is reformatting unrelated code spots.

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java [hunk 3]
```diff
@@ -29,21 +29,21 @@
 class SimpleDigest
 {
```
Comment (rfscholte) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:30]: I know this hasn't changed, but `SimpleDigest` doesn't reflect it functionality. With the extra options it makes sense to improve the javadoc saying it searches for the strongest available digest.

Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:30]: Will add, but also note that this class is package private in internal impl.

Comment (michael-o) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/SimpleDigest.java:31]: Done.

**Comment by jira-importer** [comment]
Resolve #789

**Comment by jira-importer** [comment]
Resolve #789
